service/model_manager/model_installer.py

```diff
@@ -56,26 +56,60 @@ def parse_wget_output(output):
     match = re.search(r'\d+%|\d+K \d+Kb/s', output)
     return match.group(0) if match else ""
 
-from concurrent.futures import ThreadPoolExecutor
-import asyncio
+import queue
+import threading
+
+# Create a Queue object
+download_queue = queue.Queue()
+# Create a thread that will execute the download tasks
+download_thread = None
+# Create a Condition object to share the download progress
+download_progress = threading.Condition()
+download_progress.value = {'progress': "", 'error': None}
+
+def download_worker():
+    while not download_queue.empty():
+        # Get a download task from the queue
+        (url, save_path) = download_queue.get()
+
+        # Execute the download task and update the download progress
+        download_url_with_agent(url, save_path)
+
+        # Indicate that the task is done
+        download_queue.task_done()
+
+    # If all tasks are done, reset the progress and exit
+    with download_progress:
+        download_progress.value['progress'] = ""
+        download_progress.notify_all()
```

Comment:
once all task down, do we need to close or clear the thread?

Reply:
thread will be dead if the threading function exits, no need to clean